### ISSUE TITLE
Feature Request: an additional, alternate version of HasPostCombatPromotions

### ISSUE THREAD

**Issue by Tekamthi** [LoneGazebo/Community-Patch-DLL#11430]
This one might be easy, and unlock a lot of modding potential.

Please consider adding a version of HasPostCombatPromotions that is not lost on combat... ie it functions just as the existing field and provides a new promo from the defined list, but does not disappear.  Combined with the new plague functionality we will have very powerful promo tools for modding and further development of mainbranch

**Comment by azum4roll** [comment]
Doesn't that mean it'll keep getting more promotions from every combat?

**Comment by Tekamthi** [comment]
yes, it would.... but depending on what that promo is, it could make a lot of sense this way.  When we start to explore some of the possibilities with the tables etc. added recently, this mechanism could play off the others for some depth and complexity to abilities

for example, on a very simple level:  i want a unit that cannot attack on the turn(s) immediately after its been in combat -- so i could have a dummy promo with no other abilities than this new theorized haspostcombat, and the postcombat promo itself would have canonlydefend, as well as duration of 1 turn... every time the unit is in combat, a fresh copy of the postcombat would be applied, accomplishing the desired effect.

alternatively (this one i've implemented a lot, albeit using lua helpers to re-add), say we want a unit that can travel quickly out-of-combat, but for combat balance cannot move so quickly... here we could use this to apply an after combat -x moves for y turn duration.

whenever i posted this the other day i had a few more complex theorycrafts in-mind, combined with location- and plague-promos.  maybe i am asking for a 'SelfPlague' instead?  either/or would work, idc how it functions behind the scenes, just having this modding functionality

**Comment by Tekamthi** [comment]
here's the immediate premise i would use this for, once the attack plagues and defend plagues are online in the tables (or are they now?):

recon combat role modded to more of a harasser, using attack plagues to project a significant portion of recon combat power rather than directly via cs -- to counter it, attack it before it attacks you
